tests/vm_test.go

```diff
@@ -978,6 +979,49 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			})
 		})
 	})
+
+	Context("with oc/kubectl", func() {
+		var vm *v1.VirtualMachine
+		var err error
+		var vmJson string
+
+		// Get proper commandline client: oc or kubectl
+		k8sClient := tests.GetK8sCmdClient()
+
+		AfterEach(func() {
+			if vmJson != "" {
+				err = os.Remove(vmJson)
+				Expect(err).ToNot(HaveOccurred())
+				vmJson = ""
+			}
+		})
+
+		It("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]should create VM only once", func() {
+			vm = tests.NewRandomVMWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskAlpine))
+
+			vmJson, err = tests.GenerateVMJson(vm)
+			Expect(err).ToNot(HaveOccurred(), "Cannot generate vmJson")
+
+			By("Creating VM with DataVolumeTemplate entry with k8s client binary")
+			_, _, err = tests.RunCommand(k8sClient, "create", "-f", vmJson)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verifying VM is created")
+			Eventually(func() string {
+				newVM, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &v12.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
This block appears wrong. The way this is written, the `Expect(err).ToNot(HaveOccurred())` will halt the test if it ever fails (e.g. NotFoundError). I don't think you need the `Eventually` here. We're not waiting for some event in the VM's lifecycle--we're just trying to fetch the record.

Reply:
You are right. The record is there. No need for waiting.